app/modules/parsing/utils/cache_diagnostics.py

```
@@ -0,0 +1,144 @@
+            else:
+                diagnostics["cache_misses"] += 1
+                if len(diagnostics["examples"]["cache_miss"]) < 5:
+                    diagnostics["examples"]["cache_miss"].append(
+                        {
+                            "node_id": str(node_id)[:16] if node_id else "unknown",
+                            "has_type": bool(node_type),
+                            "hash": content_hash[:12] if content_hash else "none",
+                            "text_preview": str(node_text)[:50] if node_text else "",
+                        }
+                    )
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)